### ISSUE TITLE
 Option for turning on Power Saving Mode when Battery Saver is on.

### ISSUE THREAD

**Issue by salad612** [organicmaps/organicmaps#6226]
**Is your feature request related to a problem? Please describe.**
Sometimes I want to turn on power saving mode so that it uses less battery, but I currently have to do it manually in the app settings every time since I don't want to have it always on, and my battery is not always low (also I don't know exactly what percentage the setting "when battery is low" activates at but I usually don't let my battery go sub 30%).

**Describe the solution you would like**
An additional option in the power saving mode setting that would automatically turn it on when battery saver (on android) or low power mode (on ios) is on. 

I think it makes sense because battery saver being on could indicate that the user wants to save as much battery as possible. When it is enabled the mobile OS automatically adjusts things that could result in the device being slower but more power efficient, so the option also making the organic maps app more power efficient automatically when it is on would be a nice addition.

Additionally, the user can set battery saver to turn on automatically at a certain percentage in android's settings which would then hypothetically activate the power saver mode in the app, so the effects of the current "when battery is low" option in the app would be provided anyway (although I would keep that option alongside this hypothetical new option and the existing "never" and "always" because people might still want it.)

**Describe alternatives you have considered**
- How do you solve this issue now with Organic Maps or other apps?
I currently turn power saving mode on manually in the app when I need to save power.
- Attach any examples, screenshots, or screen recordings from other apps that help us to better understand the idea.

**Additional context**

**Comment by biodranik** [comment]
Most users enable their system battery saving setting permanently. If you enable it in OM, you won't see 3D buildings, for example.

**Comment by Misalf-git** [comment]
> Most users enable their system battery saving setting permanently. If you enable it in OM, you won't see 3D buildings, for example.

So what? Isn't that exactly what those users would want to achieve?
I have no idea how you know what most users do with their system, but turn it around and they can disable power saving in OM to see 3D buildings, if they want to waste power only through OM.
Also, when or how does OM's power saving actually kick in?

**Comment by fizwidget** [comment]
> Most users enable their system battery saving setting permanently.

I don’t think this is accurate, at least not on iOS. “Low power mode” on iOS automatically disables when the battery is charged above 80%. It’s not even possible to permanently enable it. 

IMHO Organic Maps should respect the OS-level power settings, rather than have its own separate setting. The current behaviour is just unnecessary complexity in my opinion, akin to an app having its own “dark mode” preference that’s separate from the OS-level one.

**Comment by biodranik** [comment]
Of course, it is possible to [enable Low Power Mode permanently](https://www.telstra.com.au/exchange/how-to-enable-low-power-mode-automatically) on iOS.

On Android, most users enable it permanently.

Let's focus on a problem that you're trying to solve. What are the benefits of changing the behavior? How big would be the possible battery saving or loss? How to educate users that 3D buildings and some other features may unexpectedly disappear for them (and how to avoid a storm of issues/support emails from users in this case)?

Organic Maps does not consume much battery, compared to most other apps. Enabling battery saving in OM manually may save some additional bits, but it's better to measure and know the numbers (that may be different if you're using OM in a city or in the forest).

Does it make sense to measure how big battery savings you get in OM with that setting enabled and disabled in your case first, to have at least a rough understanding of the importance of the issue?

**Comment by fizwidget** [comment]
Agreed, some rigorous measurements of the impact on battery life would help with prioritisation. I didn’t mean to imply this was a particularly urgent issue, just something I thought could be improved 🙂

> What are the benefits of changing the behavior?

The benefits I see are:
* Users can toggle a single setting, rather than having to toggle settings in multiple places. 
* Less error-prone. With the current approach, users might forget to toggle OM’s setting (or never discover it in the first place). 

In my experience iOS’s “low power mode” is something people enable when they’re worried they might run out of power (perhaps on their own initiative, or perhaps in response to iOS suggesting it). As you pointed out it is possible to permanently enable it with an automation, but I suspect only a small minority of users would do that. 

I tend to use low power mode in circumstances like this:
* I’m on a hike and want to stop all unnecessary battery drain, as running out could impact my ability to navigate. 
* I’m on a long car/bus/train/plane trip and don’t have access to a charger. 
* There’s been a power outage in my area and I don’t know when it will be restored. 
* I simply used my phone more than usual, and am worried it’ll run out of power before the evening. 

In all cases I want a quick & simple way to stop unnecessary battery drain. My expectation is that UX may suffer a bit (e.g. animations disabled) in favour of battery preservation. 

Perhaps one solution would be to replace OM’s “When battery is low” option with a “Match iOS preference” one? The explanatory text below could specifically mention 3D buildings, if confusion about that is a concern.

**Comment by timonoj** [comment]
If it helps: on normal days i won't turn on battery saver mode on my phone, as i can predictably charge it. However when i travel (frequently) i have no idea when I'm going to be able to charge, and so I enable battery saving mode for extra mileage. Having to enable it separately in each app makes it unnecessarily cumbersome, and i agree that Organic Maps should just follow the OS setting.
